astropy/coordinates/spectral_coordinate.py

```diff
@@ -799,11 +671,16 @@ def with_radial_velocity_shift(self, target_shift=None, observer_shift=None):
         #  assume it's a redshift float value and convert to velocity
 
         if target_shift is None:
+            if self._observer is None or self._target is None:
+                return self.replicate()
```

Comment:
Is there a particular reason we still replicate instead of just returning `self`?

Reply:
Re: just returning self -- Then it won't be a copy anymore, like the docstring claims?

Reply:
Yep - I would always expect replicate to return a copy

Reply:
Right, of course -- `replicate` _should_ return a copy. My main question is: why are we replicating in this case? If a user gets here, they're not quite using the method as intended _and_ they're duplicating data which gives the impression that ✨ something happened ✨, but really we internally deal with this as "no observer or target; can't do anything; returning". Perhaps a warning to make the user aware, or an error to make it clear that this method _does_ require arguments to work properly?

Overall, a minor quibble, so no need to address this with any earnest. 😁

Reply:
If they not supposed to get to that part of the code, then throwing an exception would be the safest?

Reply:
@astrofrog - no conclusion seems to have been reached here...

Reply:
There could be programmatic reasons why this method gets called with no arguments (e.g if iterating over many objects and using some logic that means that sometimes it is called with no arguments) - in this case I would always want the results to be new objects, not self sometimes and a new object other times. But maybe I can make this explicit in the docstring.

Reply:
See 3465d53abe7b66922cc55848d91419a3457e12fb for a clarification in the docstring. Is this ok now?

Reply:
Looks good to me